test/attr/attr_specialize.swift

```diff
@@ -85,7 +85,7 @@ struct FloatElement : HasElt {
   typealias Element = Float
 }
 @_specialize(where T == FloatElement)
-@_specialize(where T == IntElement) // expected-error{{generic parameter 'T.Element' cannot be equal to both 'Float' and 'IntElement.Element' (aka 'Int')}}
+@_specialize(where T == IntElement) // expected-error{{generic parameter 'T.Element' cannot be equal to both 'Float' and 'Int'}}
```

Comment:
Is it possible to produce the diagnostics using the sugared types? I.e. `generic parameter 'T.Element' cannot be equal to both 'Float' and 'IntElement.Element' (aka 'Int')`, because it makes it easier for the user to understand the error message